kubernetes/windows/main.ps1

```diff
@@ -120,9 +120,56 @@ function Set-EnvironmentVariables {
     }
 
     # Set environment variable for TELEMETRY_APPLICATIONINSIGHTS_KEY
-    $aiKey = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($env:APPLICATIONINSIGHTS_AUTH))
-    [System.Environment]::SetEnvironmentVariable("TELEMETRY_APPLICATIONINSIGHTS_KEY", $aiKey, "Process")
-    [System.Environment]::SetEnvironmentVariable("TELEMETRY_APPLICATIONINSIGHTS_KEY", $aiKey, "Machine")
+    # (or fetch a custom telemetry key if set)
+    $aiKeyPassed = [System.Environment]::GetEnvironmentVariable('APPLICATIONINSIGHTS_AUTH')
+    if ( $aiKeyPassed -match '^[A-Za-z0-9=]+$') {
+        # instrumentation key directly passed, set environment variable
+        $aikey = $aiKeyPassed;
+    }
+    else {
+        $aiKeyFetched = ""
+        
+        # retry up to 5 times
+        for( $i = 1; $i -le 5; $i++) {
+            try {
+                $response = Invoke-WebRequest -uri $aiKeyPassed -UseBasicParsing -TimeoutSec 5 -ErrorAction:Stop
```

Comment:
-ErrorAction:stop  should stop the execution in case of any error and basically, container will be crashloop back state in case of any error. I presume, we want to proceed but without telemetry and in that case, we should remove "-ErrorAction:stop" proceed.